x/marker/types/msg_test.go

```diff
@@ -689,3 +689,94 @@ func TestMsgSetAccountDataRequestGetSigners(t *testing.T) {
 		require.PanicsWithError(t, "decoding bech32 failed: invalid separator index -1", testFunc, "GetSigners")
 	})
 }
+
+func TestMsgUpdateSendDenyListRequest(t *testing.T) {
+	addr := sdk.AccAddress("addr________________").String()
+	denom := "somedenom"
+	addAddr := sdk.AccAddress("addAddr________________").String()
+	removeAddr := sdk.AccAddress("removeAddr________________").String()
+
+	tests := []struct {
+		name          string
+		msg           MsgUpdateSendDenyListRequest
+		expectedError string
+	}{
+		{
+			name: "should succeed",
+			msg:  MsgUpdateSendDenyListRequest{Denom: denom, RemoveDeniedAddresses: []string{removeAddr}, AddDeniedAddresses: []string{addAddr}, Authority: addr},
+		},
+		{
+			name:          "should fail, invalid authority address",
+			msg:           MsgUpdateSendDenyListRequest{Denom: denom, RemoveDeniedAddresses: []string{removeAddr}, AddDeniedAddresses: []string{addAddr}, Authority: "invalid-address"},
+			expectedError: "decoding bech32 failed: invalid separator index -1",
+		},
+		{
+			name:          "should fail, both add and remove list are empty",
+			msg:           MsgUpdateSendDenyListRequest{Denom: denom, RemoveDeniedAddresses: []string{}, AddDeniedAddresses: []string{}, Authority: addr},
+			expectedError: "both add and remove lists cannot be empty",
+		},
+		{
+			name:          "should fail, invalid authority address",
+			msg:           MsgUpdateSendDenyListRequest{Denom: denom, RemoveDeniedAddresses: []string{removeAddr}, AddDeniedAddresses: []string{addAddr}, Authority: "invalid-address"},
+			expectedError: "decoding bech32 failed: invalid separator index -1",
+		},
+		{
+			name:          "should fail, invalid remove address",
+			msg:           MsgUpdateSendDenyListRequest{Denom: denom, RemoveDeniedAddresses: []string{"invalid-address"}, AddDeniedAddresses: []string{}, Authority: addr},
+			expectedError: "decoding bech32 failed: invalid separator index -1",
+		},
+		{
+			name:          "should succeed, invalid add address",
+			msg:           MsgUpdateSendDenyListRequest{Denom: denom, RemoveDeniedAddresses: []string{removeAddr}, AddDeniedAddresses: []string{"invalid-addrs"}, Authority: addr},
+			expectedError: "decoding bech32 failed: invalid separator index -1",
+		},
```

Comment:
"should succeed" is not correct.

I'm really not a fan of including "should succeed" or "should fail" in the test case names. It's too easy to accidentally get it wrong, or forget to correct it with a change that flips it. Plus, if the test case fails, the expected "should fail" or "should succeed" concept is conveyed by the different assertion failure messages. When looking at the test case definition, whether its expected to fail is apparent with the presence or absence of an expected error.